src/main/java/org/spongepowered/common/event/tracking/phase/packet/PacketPhaseUtil.java

```diff
@@ -165,8 +165,16 @@ public static boolean allTransactionsInvalid(final List<SlotTransaction> slotTra
     @SuppressWarnings({"rawtypes", "unchecked"})
     public static void onProcessPacket(final Packet packetIn, final INetHandler netHandler) {
         if (netHandler instanceof NetHandlerPlayServer) {
+            EntityPlayerMP packetPlayer = ((NetHandlerPlayServer) netHandler).player;
```

Comment:
```suggestion
        if (!((NetHandlerPlayServer) netHandler).player.isEntityAlive() 
           || (!(packetIn instanceof CPacketClientStatus) 
               || ((CPacketClientStatus) packetIn).getStatus() != CPacketClientStatus.State.PERFORM_RESPAWN))) {
           return;        
        }
```

Pretty sure this'll make the whole lazy evaluation of the case where the player is dead, and the packet is a respawn packet, it'll continue. if the player is dead and the packet is not a respawn packet, it'll return.

Reply:
The reason the packetPlayer variable is there is because I moved it out of the try-with-resources so it wasn't grabbing it twice. It's later passed to the StackFrame
https://github.com/SpongePowered/SpongeCommon/blob/f730075b39e8a22bad75391a1e948fef66333042/src/main/java/org/spongepowered/common/event/tracking/phase/packet/PacketPhaseUtil.java#L168-L178

Reply:
I've collapsed the if statement as you've shown with the exception of the packetPlayer variable being removed as it's required later in the code.